pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -353,7 +353,32 @@ func adsz(w http.ResponseWriter, req *http.Request) {
 		fmt.Fprintf(w, "Pushed to %d servers", len(adsClients))
 		return
 	}
+	proxyID := req.Form.Get("proxyID")
+
 	adsClientsMutex.RLock()
+	defer adsClientsMutex.RUnlock()
```

Comment:
For debug is ok - but in general it's a good practice to not do Write while holding a lock.